src/NHibernate/Proxy/AbstractProxyFactory.cs

```diff
@@ -25,6 +27,21 @@ protected bool IsClassProxy
 			get { return Interfaces.Length == 1; }
 		}
 
+		protected AbstractProxyFactory()
+		{
+		}
+
+		protected AbstractProxyFactory(SerializationInfo info, StreamingContext context)
+		{
+			EntityName = (string) info.GetValue(nameof(EntityName), typeof(string));
+			PersistentClass = (System.Type) info.GetValue(nameof(PersistentClass), typeof(System.Type));
+			Interfaces = (System.Type[]) info.GetValue(nameof(Interfaces), typeof(System.Type[]));
```

Comment:
I don’t think that proxies would even work in (all) netstandard as Reflection.Emit is a netstandard 1.x + platforms rather than a pure netstandard. 

#1425 would not really help here as I’m seraializing a `MethodInfo` as well.

Reply:
`MethodInfo` in covered in #1425.  I don't think NHibernate on .netstandard works without things like DynamicProxies.  The target isn't iOS (which requires static compilation), but the targets able to run .Net Core.

Reply:
> Update: Is this pull request going to provide a compile-time proxy generation ability?

It looks to me as being the purpose of https://github.com/nhibernate/NHibernate.ProxyGenerators. I do not expect this to change. Moreover Alexander [had mentioned](https://github.com/nhibernate/nhibernate-core/pull/1451#issuecomment-346935633) this project will need an update after this PR.

Reply:
I have plans to include proxy generators into the core (as a tool + msbuild targets) to provide support for compile time proxies for the platforms which do not support runtime proxies. But not in this PR, and, most likely, not in this milestone.